cameo/flux_analysis/analysis.py

```diff
@@ -367,6 +379,100 @@ class _PhenotypicPhasePlaneChunkEvaluator(object):
     def __init__(self, model, variable_reactions):
         self.model = model
         self.variable_reactions = variable_reactions
+        objective_reactions = [reaction for reaction in model.reactions
+                               if reaction.objective_coefficient != 0]
+        if len(objective_reactions) != 1:
+            raise Exception('multiple objectives not supported')
+        self.product_reaction = objective_reactions[0]
+        self.c_source_reactions = self.get_c_source_reactions()
+
+    def get_c_source_reactions(self):
+        """ carbon source reactions
+
+        Returns
+        -------
+        list
+           list of media reaction currently with negative flux or 0 if non defined
+        """
+        medium_reactions = [self.model.reactions.get_by_id(reaction) for reaction in self.model.medium.reaction_id]
+        self.model.solve()
+        source_reactions = [reaction for reaction in medium_reactions if
+                            (reaction.n_carbon > 0) and (reaction.flux < 0)]
```

Comment:
For many reasons concerning safe usage you should not use *reaction.flux*.
The solution object generated <code>_production_envelope_inner</code> should be an argument of all of this methods you implemented.
The *solution* object contains the *.fluxes* attribute that will contain the fluxes calculated. By doing that you don't need to concern about catching the Exceptions raised during optimizing the model.

The code in  <code>_production_envelope_inner</code> becomes something like this.

<code>

    self.model.objective.direction = 'min'

    try:

        solution = self.model.solve().f

    try:

        solution = self.model.solve()
        interval.append(solution.f)

        interval_carbon_yield.append(self.carbon_yield(solution))

        interval_mass_yield.append(self.mass_yield(solution))

    except (Infeasible, UndefinedSolution):

        interval.append(0)

        interval_carbon_yield.append(0)

        interval_mass_yield.append(0)
    self.model.objective.direction = 'max'
    try:

        solution = self.model.solve()

        interval.append(solution.f)

        interval_carbon_yield.append(self.carbon_yield(solution))

        interval_mass_yield.append(self.mass_yield(solution))

    except (Infeasible, UndefinedSolution):

        interval.append(0)

        interval_carbon_yield.append(0)

        interval_mass_yield.append(0)

</code>